tools/lint/lint.py

```diff
@@ -624,6 +624,32 @@ def check_script_metadata(repo_root, path, f):
     return errors
 
 
+def check_multiple_identical_values(repo_root, paths):
+    """
+    Runs lints that check for multiple identical paths
+    in all_paths_lints.
+
+    :param repo_root: the repository root
+    :param paths: a list of all the paths within the repository
+    :returns: a list of errors found in ``path``
+    """
+
+    file_dict = dict()
+    errors = []
+    for paths in paths:
+        file, file_extension =  os.path.splitext(paths)
+        if file not in file_dict.keys():
```

Comment:
There's a type in the standard library called `defaultdict` that encapsulates this check-and-create behaviour; let' use that.